composer/__init__.py

```diff
@@ -1,14 +1,5 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
-from composer import algorithms as algorithms
```

Comment:
Having these re-exports here broke the `from composer import *` in the doctest fixtures

Reply:
Random thought - should we have a doctest here to test for `from composer import ____` imports that we expect to work (that way if someone changes something within a module like callbacks that somehow breaks importing directly from Composer we catch it here)

Reply:
(pasting reply from the conversation)

Any import from a submodule will attempt to load the parent module's __init__.py, so our tests already check to ensure that the parent's __init__.py is bug-free. As for ensuring that, say, composer.callbacks doesn't have a bug in it, we should use the highest-level import whenever possible (e.g. from composer import Trainer, not from composer.trainer.trainer import Trainer) in the tests. But going to leave that out of this PR as that would be a larger change.